components/tidb_query_vec_expr/src/impl_string.rs

```diff
@@ -28,6 +29,26 @@ pub fn length(arg: &Option<Bytes>) -> Result<Option<i64>> {
     Ok(arg.as_ref().map(|bytes| bytes.len() as i64))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn unhex(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    if let Some(content) = arg {
+        // hex::decode will fail on odd-length content
+        // but mysql won't
+        // so do some padding
+        let padded_content = if content.len() % 2 == 1 {
+            let mut result = b"0".to_vec();
+            result.extend_from_slice(content);
+            result
+        } else {
+            content.clone()
+        };
+        Ok(hex::decode(padded_content).ok())
```

Comment:
I think it's better to propagate the error instead of silently ignore it.

Reply:
> I think it's better to propagate the error instead of silently ignore it.

Can you tell me which kind of Error should it be?
Is it an `EvaluateError::Other` or should I add another Error type?

Reply:
Oh sorry, I was wrong. It's correct to just return `None`.